Allow archived projects to show in search

wafflecoffee opened the pull request:
Resolves MOD-96

Geometrically reviewed (changes requested):
SQL diff is also pretty large in this one. Could you reset it to the one in HEAD and just run `cargo sqlx prepare` from there?

wafflecoffee commented:
@Geometrically It's identical.
```sh
emma ~/repos/modrinth/labrinth/ (archived-in-search) $ git co master
Switched to branch 'master'
Your branch is up to date with 'upstream/master'.
emma ~/repos/modrinth/labrinth/ (master) $ git pull
Already up to date.
emma ~/repos/modrinth/labrinth/ (master) $ cp sqlx-data.json /tmp
emma ~/repos/modrinth/labrinth/ (master) $ git co archived-in-search 
Switched to branch 'archived-in-search'
Your branch is up to date with 'origin/archived-in-search'.
emma ~/repos/modrinth/labrinth/ (archived-in-search) $ mv /tmp/sqlx-data.json .
emma ~/repos/modrinth/labrinth/ (archived-in-search) $ cargo sqlx prepare
    [snip]
    Finished dev [unoptimized + debuginfo] target(s) in 4m 25s
query data written to `sqlx-data.json` in the current directory; please check this into version control
emma ~/repos/modrinth/labrinth/ (archived-in-search) $ git commit sqlx-data.json 
On branch archived-in-search
Your branch is up to date with 'origin/archived-in-search'.

nothing to commit, working tree clean
```
